values/src/helpers.rs

```diff
@@ -484,41 +489,49 @@ pub fn hour_relative_minute(h: u32, m: i32, is_12_clock: bool) -> RuleResult<Tim
     hour_minute(shifter_hour, normalized_minute, is_12_clock)
 }
 
-pub fn cycle(grain: Grain) -> RuleResult<TimeValue> {
-    Ok(TimeValue::constraint(Cycle::rc(grain)).form(Form::Cycle(grain)))
+pub fn cycle(grain: Grain) -> RuleResult<DatetimeValue> {
+    Ok(DatetimeValue::constraint(Cycle::rc(grain)).form(Form::Cycle(grain)))
 }
 
-pub fn cycle_nth(grain: Grain, n: i64) -> RuleResult<TimeValue> {
-    Ok(TimeValue::constraint(Cycle::rc(grain).take_the_nth(n)).form(Form::Cycle(grain)))
+pub fn cycle_nth(grain: Grain, n: i64) -> RuleResult<DatetimeValue> {
+    Ok(DatetimeValue::constraint(Cycle::rc(grain).take_the_nth(n)).form(Form::Cycle(grain)))
 }
 
-pub fn cycle_nth_after(grain: Grain, n: i64, after_value: &TimeValue) -> RuleResult<TimeValue> {
-    Ok(TimeValue::constraint(Cycle::rc(grain).the_nth(n).after(&after_value.constraint)).form(Form::Cycle(grain)))
+pub fn cycle_nth_after(grain: Grain, n: i64, after_value: &DatetimeValue) -> RuleResult<DatetimeValue> {
+    Ok(DatetimeValue::constraint(Cycle::rc(grain).the_nth(n).after(&after_value.constraint)).form(Form::Cycle(grain)))
 }
 
 pub fn cycle_nth_after_not_immediate(grain: Grain,
                                      n: i64,
-                                     after_value: &TimeValue)
-                                     -> RuleResult<TimeValue> {
-    Ok(TimeValue::constraint(Cycle::rc(grain)
+                                     after_value: &DatetimeValue)
+                                     -> RuleResult<DatetimeValue> {
+    Ok(DatetimeValue::constraint(Cycle::rc(grain)
                                  .the_nth(n)
                                  .after_not_immediate(&after_value.constraint)).form(Form::Cycle(grain)))
 }
 
-pub fn cycle_n(grain: Grain, n: i64) -> RuleResult<TimeValue> {
-    Ok(TimeValue::constraint(Cycle::rc(grain).take(n)).form(Form::Cycle(grain)))
+pub fn cycle_n(grain: Grain, n: i64) -> RuleResult<DatetimeValue> {
+    Ok(DatetimeValue::constraint(Cycle::rc(grain).take(n)).form(Form::Cycle(grain)))
 }
 
-pub fn cycle_n_not_immediate(grain: Grain, n: i64) -> RuleResult<TimeValue> {
-    Ok(TimeValue::constraint(Cycle::rc(grain).take_not_immediate(n)).form(Form::Cycle(grain)))
+pub fn cycle_n_not_immediate(grain: Grain, n: i64) -> RuleResult<DatetimeValue> {
+    Ok(DatetimeValue::constraint(Cycle::rc(grain).take_not_immediate(n)).form(Form::Cycle(grain)))
 }
 
-pub fn easter() -> RuleResult<TimeValue> {
+pub fn grain_greather_than_day(grain: Grain) -> bool {
+    if [Grain::Week, Grain::Month, Grain::Year, Grain::Quarter].contains(&grain) {
+        true
+    } else {
+        false
+    }
+}
```

Comment:
This would better be a method of the `Grain` enum.

Reply:
✅